lightbluetent/home.py

```diff
@@ -59,34 +59,35 @@ def register():
         if moderator_pw == attendee_pw:
             errors.append("An error occured. Please try again.")
 
-        if User.query.filter_by(email=email_address):
+        if User.query.filter_by(email=email_address).first():
             errors.append("That email address is already registered.")
-        if Society.query.filter_by(uid=uid):
+        if Society.query.filter_by(uid=uid).first():
             errors.append("That society short name is already in use.")
-        elif (Society.query.filter_by(attendee_pw=attendee_pw)
-                or Society.query.filter_by(moderator_pw=moderator_pw)
-                or Society.query.filter_by(bbb_id=bbb_id)):
+        elif (Society.query.filter_by(attendee_pw=attendee_pw).first()
+                or Society.query.filter_by(moderator_pw=moderator_pw).first()
+                or Society.query.filter_by(bbb_id=bbb_id).first()):
             errors.append("An error occured. Please try again.")
 
 
         if not errors:
 
             # TODO: BBB create() API call goes here?
 
-            admin = User(email=email_address,
-                         name=name,
-                         society_id=society.id,
-                         crsid=auth_decorator.principal)
-
             society = Society(short_name=soc_short_name,
-                              name=name,
+                              name=soc_name,
                               attendee_pw=attendee_pw,
                               moderator_pw=moderator_pw,
                               uid=uid,
-                              bbb_id=bbb_id,
-                              admins=admin)
+                              bbb_id=bbb_id)
 
             db.session.add(society)
+            db.session.commit()
```

Comment:
Can we add both entries then commit the two in one go?

Reply:
Tried this, but Sqlalchemy complains about society.id being null when creating the user, I suppose because the society isn't committed to the database yet. Is there a better way of doing this?

Reply:
Ah OK, looks good then.